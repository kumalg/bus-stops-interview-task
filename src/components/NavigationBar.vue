<template>
    <BaseCard :without-paddings="true">
        <nav class="navigation-bar">
            <RouterLink
                v-for="route in ROUTE_KEYS"
                :key="route"
                :to="ROUTE_PATHS[route]"
                class="navigation-item"
            >
                {{ ROUTE_PAGE_TITLES[route] }}
            </RouterLink>
        </nav>
    </BaseCard>
</template>

<script setup lang="ts">
import { ROUTE_PATHS, ROUTE_PAGE_TITLES, ROUTE_KEYS } from '@/router/routes'

import BaseCard from '@/components/BaseCard.vue'
</script>

<style lang="scss" scoped>
.navigation-bar {
    padding: 0 1.5rem;
    height: 4rem;
    display: flex;

    .navigation-item {
        padding: 0 1.5rem;
        display: flex;
        align-items: center;
        border-bottom: 2px solid transparent;
        text-decoration: none;
        font-size: 0.875rem;
        line-height: 1.5rem;
        color: $color-gray-600;
        transition-property: border-bottom-color, color;
        transition-duration: .2s;
        transition-timing-function: ease;

        &:hover {
            color: $color-gray-700;
        }

        &.router-link-active {
            border-bottom-color: $color-accent;
            color: $color-gray-800;
        }
    }
}
</style>