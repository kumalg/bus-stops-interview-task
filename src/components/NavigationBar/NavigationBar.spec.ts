import { shallowMount } from '@vue/test-utils';

import NavigationBar from './NavigationBar.vue';

describe('NavigationBar', () => {
    const RouterLinkStub = {
        template: '<router-link-stub><slot/></router-link-stub>'
    };
    const BaseCardStub = {
        template: '<base-card-stub><slot/></base-card-stub>'
    };

    const NavigationBarFactory = (props: InstanceType<typeof NavigationBar>['$props'] = {}) => {
        return shallowMount(NavigationBar, {
            props,
            global: {
                stubs: {
                    BaseCard: BaseCardStub,
                    RouterLink: RouterLinkStub
                }
            }
        });
    };

    it('renders proper html markup', () => {
        const wrapper = NavigationBarFactory();

        expect(wrapper.element).toMatchSnapshot();
    });

    it('renders both Bus Lines and Stops links', () => {
        const wrapper = NavigationBarFactory();

        expect(wrapper.findAllComponents(RouterLinkStub).length).toBe(2);
    });

    it('renders proper Bus Lines link text and path', () => {
        const wrapper = NavigationBarFactory();
        const [link] = wrapper.findAllComponents(RouterLinkStub);

        expect(link.text()).toBe('Bus Lines');
        expect(link.attributes()['to']).toBe('/');
    });

    it('renders proper Stops link text and path', () => {
        const wrapper = NavigationBarFactory();
        const [, link] = wrapper.findAllComponents(RouterLinkStub);

        expect(link.text()).toBe('Stops');
        expect(link.attributes()['to']).toBe('/stops');
    });
});
