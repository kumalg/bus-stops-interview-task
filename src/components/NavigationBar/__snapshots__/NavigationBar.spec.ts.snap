// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders proper html markup 1`] = `
<base-card-stub
  without-paddings="true"
>
  
  <nav
    class="navigation-bar"
  >
    
    <router-link-stub
      class="navigation-item"
      to="/"
    >
      
      Bus Lines
      
    </router-link-stub>
    <router-link-stub
      class="navigation-item"
      to="/stops"
    >
      
      Stops
      
    </router-link-stub>
    
  </nav>
  
</base-card-stub>
`;
